utils/SimpleJson.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <assert.h>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace utils
+{
+
+class SimpleJson
+{
+public:
+    enum Type
```

Comment:
The type is strictly not necessary. The type is mostly obvious from the start character of the SimpleJson item.
The only ambiguity that would require more processing of characters is when it is a number. We cannot tell if it is a float or integer until we have looked for decimal point. Even then it is up to the user to retrieve it as a double.

Reply:
They type is derived automatically while linear scanning, and helps to whether getter is eligible. Also, it helps to make a special SimpleJsonNone value.